app/stac_api/admin.py

```diff
@@ -120,11 +121,26 @@ class Media:
     wms_layer = 'ch.swisstopo.pixelkarte-farbe-pk1000.noscale'
     wms_url = 'https://wms.geo.admin.ch/'
     list_display = ['name', 'collection']
-    list_filter = [CollectionFilter]
+    list_filter = [CollectionFilterForItems]
 
     def get_search_results(self, request, queryset, search_term):
         queryset, use_distinct = super().get_search_results(request, queryset, search_term)
-        if search_term.startswith('"') and search_term.endswith('"'):
+
+        # The following few lines are a bit hacky and are needed for the item dropdown list
+        # to depend on the currently selected collection in the collection dropdown filter.
+        # With this "hack", only those items appear in the "filter by item name" dropdown list,
+        # that belong to the currently selected collection in the "filter by collection name"
+        # dropdown list. Otherwise all items would appear in the dropdown list, which does not
+        # make sense.
+
+        # this asserts that the request comes from the autocomplete filters.
+        if "autocomplete" in request.__dict__['environ']['PATH_INFO']:
+            if "item__collection" in request.__dict__['environ']['HTTP_REFERER']:
+                current_collection_pk = request.__dict__['environ']['HTTP_REFERER'].split(
+                    'item__collection='
+                )[1].split("&")[0]
+                queryset = self.model.objects.filter(collection__pk__exact=current_collection_pk)
```

Comment:
This queryset will serve as input for the dropdown list of the "filter by item name" filter.